src/routes/learnbyheart/+page.svelte

```diff
@@ -0,0 +1,685 @@
+<script>
+	import { SvelteComponentDev, prevent_default } from "svelte/internal"
+	import Points from './Points.svelte';
+	import { onMount } from 'svelte';
+	import { get } from "svelte/store"
+	import { goto } from "$app/navigation"
+	import { page } from "$app/stores"
+
+	let points = 0;
+	let totalAsked = 0;
+	let pointsInstance;
+	let started = false; // used to show or not show the intro text.
+	let message = '';
+	let finished = false;
+	let showPrevContext = false;
+	let retype = false;
+	let curIndex = 0; // used to know what question to ask. 
+	let lastCurIndex = 0; // used to not ask the same question twice in a row. 
+	let lastQuestion = '';
+	let lastContext = '';
+	let fase = 0;
+	let latestMistakeIndexes = [];
+	let tempMessage = 'Reading the data';
+	let startIndex = 1;
+  	let stopIndex = 10;
+	// just initialize, not used, later set by a function. 
+	let startIndexOptions = [-1,-2,-3];
+	let stopIndexOptions = [-4,-5,-6];
+
+	// Used for the user input, also to display it. 
+	let userInput = '';
+	let value = '';
+	let count = 1;
+	let pageFase = 0;
+	let goToNextQuestion = false;
+	let notYetAsked = [];
+	let alreadyAsked = [];
+	let weights = []; 
+	let nrOfZeros = 0;
+	const nrOfNewQuestions = 5; // number of NEW questions in a streak, after that it reaskes the latest mistakes
+	const newQuestionsMessage = 'Now asking ' + nrOfNewQuestions.toString() +  ' or less new questions'
+	let datasetsHaveLoaded = false;
+	// The file is at /static/learn.. but in svelte the static folder contents are available without having to start with /static
+	// For the capitals, 
+	// Files obtained using a combination of https://www.enchantedlearning.com/geography/capitals/bycontinent.shtml
+	// And cleaning it up with ChatGPT. 
+	let pathToData = 'learnbyheart_Data/aifacts.txt'
+
+	let endMessage = 'You are at the end, the only option now is to hit restart or refresh the page for the same effect'
+
+	// This is a bit of an ugly hack, the server loads the page once before the browser has run the onMount function.
+	// In the html it loads dataset[0].question so here I give it some nonsense to load once that the user sees for less than one second.
+	let dataset = [{question: 'Loading', answer: 'nonsense', weight: -1, context:'Cool context here'}];
+	let datasets = [];
+	let datasetNames = [];
+	let datasetNamesAndIndexes =[];
+
+	// This onMount function is needed to run this loading of a file operation in the browswer, because the server did not allow a 
+	// fetch operation. The text file contains multiple subsets with lines: question;answer;context
+	onMount(async () => {
+	try {
+		const response = await fetch(pathToData); 
```

Comment:
This will not work with server-side rendering, you can't use fetch here, needs to be in the load function.

https://kit.svelte.dev/docs/load